lucille-core/src/main/java/com/kmwllc/lucille/core/Stage.java

```diff
@@ -52,7 +50,7 @@ public abstract class Stage {
   private Counter childCounter;
 
   public Stage(Config config) {
-    this(config, new StageSpec());
+    this(config, ConfigSpec.forStage());
```

Comment:
Since this paradigm is repeated in so many places, I'm thinking of a name change to make it even more concise: ConfigSpec -> Spec and forStage() -> stage()

So it would look like Spec.stage().withRequiredProperties(...

what do you think?